### ISSUE TITLE
Generating testings for the subset selection algorithms

### ISSUE THREAD

**Issue by fwmeng88** [theochem/DiverseSelector#20]
For the evaluation metrics, I don't think generating the tests is challenging. 

But when it comes to generating tests for different algorithms, things become tricky. Duplicating the original results would be good, but the original dataset is not always available. I would think of having a set of molecules with distinct structures/functional groups and see how the algorithm work. For example, we can have alkanes, alcohols, esters, ketones, *et al.*, then we can test if the implemented method is going to work or not.

Do you have any idea on how to make it work? @PaulWAyers @RichRick1 @JuansaCollins @Khaleeh @alnaba1 
Thank you!

**Comment by PaulWAyers** [comment]
This is a case where unit tests are probably most important, because at least we can be sure every small part works correctly.

The only cases I can think of beyond that are cases where the right solution is very clear. That's possible in some cases, when you compose a dataset. For example, put points (or clusters of points) on a grid. In those cases, a plot of the sample can reveal whether things are working correctly.

**Comment by fwmeng88** [comment]
Thanks for the input which is very helpful! @PaulWAyers 
Can you add a little more explanation on the idea of using a grid? Do you mean visualize the chemical diversity using some dimensionality reduction techniques, such as PCA or tSNE? Because visualizing the high-dimensional space of molecular features is not practical. Thanks! @PaulWAyers

**Comment by PaulWAyers** [comment]
I mean that if you have points at (x,y) where x=0,1,2,... and y=0,1,2,... and you select just 1/4 of the points, then max-diversity should give something like x=0,2,4,... y=0,2,4... or x=1,3,5... y=1,3,5,.... Similarly if you add a (small) cluster of points about each grid point. So there are various "intuitive graphical tests" that can be run then.

**Comment by fwmeng88** [comment]
It makes sense. Thank you!

**Comment by RichRick1** [comment]
> I mean that if you have points at (x,y) where x=0,1,2,... and y=0,1,2,... and you select just 1/4 of the points, then max-diversity should give something like x=0,2,4,... y=0,2,4... or x=1,3,5... y=1,3,5,.... Similarly if you add a (small) cluster of points about each grid point. So there are various "intuitive graphical tests" that can be run then.

In addition to this. We can just generate blobs using [sklearn function](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_blobs.html) and just make sure that output has equal-ish amount of points from every blob. In this way we can test it for more than 2 dimensions without a huge efforts
@fwmeng88

**Comment by RichRick1** [comment]
We can try to use Spectrum dataset for selecting. At this point we have set of alcohols, ketones and alcohols spectra (around 30-ish molecules of every class). In addition to this we have a metric for calculating similarity. It might be interesting to select couple of elements of  each class. The good part is we can test the way user defining it's own metric and we know the labels of the classes, so technically we know the output.
@fwmeng88 @PaulWAyers  what do you think?

**Comment by fwmeng88** [comment]
This seems a good testing dataset. Thanks!

**Comment by PaulWAyers** [comment]
I'm not sure about testing per se; it would be a great example for a notebook/paper explaining how to use the code though, and we could potentially use a more complete subset (beyond the small one we are using right now).

**Comment by fwmeng88** [comment]
This is suppressed by #48.
